### PR TITLE
#166790003 Users Should Be Able To Like And Or Dislike Articles 

### PR DESCRIPTION
**Pull request by kodek-sleuth**
#### What does this PR do?
Enables a user to like/dislike an article
#### Description of Task to be completed
- To set up like and dislike functionality
#### How should this be manually tested?
1. Run the following commands in the terminal
```
git clone https://github.com/andela/ah-92explorers-backend.git
cd ah-92explorers-backend
git checkout ft-like-unlike-article-166790003
npm install
npm start
```
2. Create a .env file and add the following snippets
```
DATABASE_URL=postgres://username:password@localhost:5432/authorshaven
DATABASE_=postgres://username:password@localhost:5432/testing-ah
SECRET=whatever you want it to be
```
3. Install `sequelize` globally
```
sudo npm install -g sequelize
```
4.  Create a database using `sequelize db:create`
5. Install and setup Postman to consume CRUD endpoints
6. Run the following requests in Postman using `localhost:3000`
## Endpoints
|  Method  |  Endpoint  |  Task  |
|  --- |  --- |  ---  |
|  `POST`  |  `/api/like/:articleSlug`  |  `like an article`  |
|  `POST`  |  `/api/dislike/:articleSlug`  |  `dislike an article`  |
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#166790003](https://www.pivotaltracker.com/story/show/166790003)
#### Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

### CODE REVIEW

**Review by elemanhillary-zz — DISMISSED** [review]
ready for merge

**Review by Hervera — DISMISSED** [review]
Ready to be approved

**Review by IsaiahRn — DISMISSED** [review]

**Review by Hervera — CHANGES_REQUESTED** [review]
Rebase from develop, `checkArticleOwner` file in the `middlewares` folder was renamed to `checkResourceOwner`

File: controllers/likes.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+          error: 'failed to find an article to like'
+        });
+      }
+      const user = await users.findOne({
+        where: { email: req.decoded.email },
+        include: [{ model: likes }]
+      });
+      const like = await likes.findOne({ where: { articleSlug: article.slug, userId: user.id }, attributes: ['typeState'] });
+
+      switch (true) {
```
Comment (elemanhillary-zz) [inline controllers/likes.js:25]: the switch `statement` and the `true`. am not understanding the use  of a switch here. On what or which condition do are you using this.

File: controllers/likes.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        case like === null:
```
Comment (elemanhillary-zz) [inline controllers/likes.js]: `!like` would do some good

File: controllers/likes.js (hunk 1 above)
Comment (elemanhillary-zz) [inline controllers/likes.js:25]: refactor the whole `switch` statement. And you dont have a `default` case.

File: controllers/likes.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+  static async dislikeArticle(req, res) {
+    try {
+      const { articleSlug } = req.params;
+      const article = await articles.findOne({
+        where: { slug: articleSlug },
+        include: [{ model: likes }]
+      });
+
+      if (article === null) {
```
Comment (elemanhillary-zz) [inline controllers/likes.js]: replace `article === null` to `!article`

File: controllers/likes.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        return res.status(404).json({
+          error: 'failed to find an article to dislike'
```
Comment (elemanhillary-zz) [inline controllers/likes.js:79]: change `failed to find an article to dislike` to `article not found`

File: controllers/likes.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+            {
+              typeState: 0
+            },
+            {
+              where: { articleSlug: article.slug }
+            }
+          );
+          return res.status(201).json({
+            message: 'successfully disliked article'
+          });
```
Comment (elemanhillary-zz) [inline controllers/likes.js:121]: i dont say the `default case statement`

File: controllers/likes.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        return res.status(404).json({
+          error: 'failed to find an article to dislike'
+        });
+      }
+      const user = await users.findOne({
+        where: { email: req.decoded.email },
+        include: [{ model: likes }]
+      });
+      const like = await likes.findOne({ where: { articleSlug: article.slug, userId: user.id }, attributes: ['typeState'] });
+      switch (true) {
```
Comment (elemanhillary-zz) [inline controllers/likes.js:87]: in which cases are you using the `switch` statement.

**Review by elemanhillary-zz — CHANGES_REQUESTED** [review]

File: controllers/likes.js (hunk 1 above)
Comment (kodek-sleuth) [reply controllers/likes.js:25]: A default case can be foregone since there is a catch block with a return at the end.

**Review by elemanhillary-zz — DISMISSED** [review]
ready for merge

**Review by niyoceles — APPROVED** [review]

**Review by niyoceles — DISMISSED** [review]

**Review by Hervera — DISMISSED** [review]
Ready to be merged

**Review by IsaiahRn — DISMISSED** [review]

**Review by niyoceles — DISMISSED** [review]

**Review by niyoceles — APPROVED** [review]

**Review by niyoceles — DISMISSED** [review]

**Review by elemanhillary-zz — CHANGES_REQUESTED** [review]
coverage dropped

**Review by elemanhillary-zz — APPROVED** [review]
ready for merge

**Review by niyoceles — APPROVED** [review]

**Review by IsaiahRn — APPROVED** [review]

**Pull request merged**